src/webui/webui.cpp

```diff
@@ -77,11 +79,18 @@ void WebUI::configure()
                 m_httpServer->close();
         }
 
-#ifndef QT_NO_OPENSSL
         if (pref->isWebUiHttpsEnabled()) {
-            const QByteArray certs = pref->getWebUiHttpsCertificate();
-            const QByteArray key = pref->getWebUiHttpsKey();
-            bool success = m_httpServer->setupHttps(certs, key);
+            const auto readData = [](const QString &path) -> QByteArray
+            {
+                QFile file(path);
+                if (!file.open(QIODevice::ReadOnly))
+                    return {};
+                return file.read(1024 * 1024);
+            };
+            const QByteArray cert = readData(pref->getWebUIHttpsCertificatePath());
```

Comment:
Shouldn't we track whether the files were changed from last loading?

Reply:
what for?

Reply:
To prevent reloading of the same data. Just small optimization. No big deal.

Reply:
let's keep it simple for now.